hardware-security-module/README.md

```diff
@@ -0,0 +1,127 @@
+# Fabric Gateway HSM Samples
+
+The samples show how to create client applications that invoke transactions with HSM Identities using the
+new embedded Gateway in Fabric.
+
+The samples will only run against Fabric v2.4 and higher.
+
+Sample client applications are available to demonstrate the features of the Fabric Gateway and associated SDKs using this network.
+
+## Installations
+
+### C Compilers
+
+In order for the client application to run successfully you must ensure you have C compilers and Python 3 (Note that Python 2 may still work however Python 2 is out of support and could stop working in the future) installed otherwise the node dependency `pkcs11js` will not be built and the application will fail. The failure will have an error such as
+
+```
+Error: Cannot find module 'pkcs11js'
+```
+
+how to install the required C Compilers and Python will depend on your operating system and version.
+
+### SoftHSM
+
+In order to run the application in the absence of a real HSM, a software
+emulator of the PKCS#11 interface is required.
+For more information please refer to [SoftHSM](https://www.opendnssec.org/softhsm/).
+
+SoftHSM can either be installed using the package manager for your host system:
+
+* Ubuntu: `sudo apt install softhsm2`
+* macOS: `brew install softhsm`
+* Windows: **unsupported**
+
+Or compiled and installed from source:
+
+1. install openssl 1.0.0+ or botan 1.10.0+
+2. download the source code from <https://dist.opendnssec.org/source/softhsm-2.5.0.tar.gz>
+3. `tar -xvf softhsm-2.5.0.tar.gz`
+4. `cd softhsm-2.5.0`
+5. `./configure --disable-gost` (would require additional libraries, turn it off unless you need 'gost' algorithm support for the Russian market)
+6. `make`
+7. `sudo make install`
+
+### PKCS#11 enabled fabric-ca-client binary
+To be able to register and enroll identities using an HSM you need a PKCS#11 enabled version of `fabric-ca-client`
+To install this use the following command
+
+```bash
+go install -tags 'pkcs11' github.com/hyperledger/fabric-ca/cmd/fabric-ca-client@latest
```

Comment:
This was my only stumbling block when reviewing the PR.  I had tried running with the default `fabric-ca-client` 1.5.5 binaries, installed with the bootstrap.sh installer to a local scratch / bin directory.  Either add a note here that the client binary installed to GOROOT with `-tags` .... should be added to the PATH, or change `generate-hsm-user.sh` script to explicitly call the installed client from GOROOT.

Reply:
GOROOT is automatically added to the  PATH right ?